cpp-matrix/index.js

```diff
@@ -612,8 +612,16 @@ function generateMatrix(compilerVersions, standards, max_standards, latest_facto
                     entry['container'] = 'ubuntu:18.04'
                 }
             } else if (compiler === 'clang') {
-                if (semver.satisfies(minSubrangeVersion, '>=12')) {
-                    entry['runs-on'] = 'ubuntu-22.04'
+                if (semver.satisfies(minSubrangeVersion, '=13 || =14')) {
```

Comment:
This can be

```
                  if (semver.satisfies(minSubrangeVersion, '>= 16')) {
                    entry['runs-on']   = 'ubuntu-22.04'
                    entry['container'] = 'ubuntu:23.04'
                  } else if (semver.satisfies(minSubrangeVersion, '>= 12')) {
                      entry['runs-on'] = 'ubuntu-22.04'
                      entry['container'] = 'ubuntu:22.04'
                  } else if (semver.satisfies(minSubrangeVersion, '>= 6')) {
                      entry['runs-on'] = 'ubuntu-22.04'
                      entry['container'] = 'ubuntu-20.04'
                  } else if ...
```

This is already simplifying the redundant `=13 | =14` and applying Peter's suggestion to always use a container for clang. I hope the recommended install packages don't change.